src/pages/Company.js

```diff
@@ -0,0 +1,234 @@
+import * as IMG from '../util/Image.js';
+import './css/Company.css';
+
+export default function Company() {
+  const companyName = useRef();
+  const companyAddress = useRef();
+  const fileInput = useRef(null);
+  const [showAlertPic, setShowAlertPic] = useState(false);
+  const [company, setCompany] = useState({});
+  const maxPhotoSize = 1024 * 1024 * 6; //KB
+
+  useEffect(() => {
+    async function getData() {
+      try {
+        const response = await Request(
+          'https://api.bandla.uz/api/user-panel/company/find',
+          'get',
+          null,
+          null,
+          true,
+          null,
+          navigateToLogin
+        );
+
+        setCompany(response);
+      } catch (error) {
+        if (error.response?.data?.errors) {
+          toast.error(error.response.data.errors[0]);
+        } else {
+          toast.error("Kompaniya yuklashda xatolik ro'y berdi");
+        }
+      }
+    }
+    getData();
+  }, []);
+
+  const navigateToLogin = () => {
+    window.location.href = 'login';
+  };
+
+  async function fileInputChange() {
+    setShowAlertPic(true);
+  }
+
+  function hideAlertPic() {
+    setShowAlertPic(false);
+    fileInput.current.value = '';
+  }
+
+  const changePhoto = async (event) => {
+    const file = event.files[0];
+    if (file.size > maxPhotoSize) {
+      toast.error("Photo surati hajmi 6 MB dan kam bo'lishi kerak!");
+      return;
+    }
+    try {
+      const formData = new FormData();
+      formData.append('file', file);
+      const response = await Request(
+        'https://api.bandla.uz/api/file-store',
+        'post',
+        null,
+        formData,
+        true,
+        navigateToLogin
+      );
+
+      setShowAlertPic(false);
+      setCompany(response);
+    } catch (error) {
+      if (error.response.data?.errors) {
+        toast.error(error.response.data.errors[0]);
+      } else {
+        toast.error("Photo suratni yuklashda xatolik ro'y berdi");
+      }
+    }
+  };
+
+  async function setSizeToImg(e) {
+    const img = e.target;
+    const parent = img.parentElement;
+
+    const imgMaxSize = (await IMG.getData(fileInput.current.files[0])).size.max;
+    const parentMinSize = IMG.minSize(parent.clientWidth, parent.clientHeight);
+
+    if (imgMaxSize === 'square') {
+      img.style[parentMinSize] = '100%';
+      parent.style[parentMinSize] = '400px';
+    }
+    if (imgMaxSize !== 'square') {
+      img.style[imgMaxSize] = '100%';
+      parent.style[imgMaxSize] = '400px';
+    }
+  }
+
+  async function createCompany() {
+    const companyData = {
+      name: companyName.current.value,
+      address: companyAddress.current.value,
+      photoUrl: company?.data?.data?.url,
+    };
+
+    try {
+      const response = await Request(
+        'https://api.bandla.uz/api/user-panel/company/create',
+        'post',
+        null,
+        companyData,
+        true,
+        navigateToLogin
+      );
+
+      setCompany(response);
```

Comment:
xato setCopmany ichiga faqat company malutmoalrini berish kere hozi resonse ichida request jonatishdagi har xil balo battar musr malumotla bor
```suggestion
      setCompany(response.data.data);
```